File: ruby/lib/casilla.rb
require_relative('sorpresa')
require_relative('mazo_sorpresas')
require_relative('jugador')
require_relative('titulo_propiedad')

module Civitas
    class Casilla
        def initialize(nombre)
            @nombre=nombre
        end
        attr_reader :nombre

        def jugadorCorrecto(actual, todos)
            return actual>=0&&actual<todos.length
        end

        def informe(actual, todos)
            if (jugadorCorrecto(actual, todos)) 
                Diario.instance.ocurreEvento(todos[actual].nombre + ": Casilla " + toString())
            end
        end

        def recibeJugador(actual, todos)
        end

        def toString
            return "\nCasilla: " + @nombre
        end
    end
end

File: java/src/civitas/Casilla.java
package civitas;
import java.util.ArrayList;

public class Casilla {
    protected String nombre;

    public Casilla() {
        nombre = "";
    }

    public Casilla(String nombre) {
        this.nombre = nombre;
    }

    public String getNombre() {
        return nombre;
    }

    private void informe(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            Diario.getInstance().ocurreEvento(todos.get(actual).getNombre() + ": Casilla " + toString());
        }
    }

    public boolean jugadorCorrecto(int actual, ArrayList<Jugador> todos) {
        return actual >= 0 && actual < todos.size();
    }

    public String toString() {
        return "Casilla: " + nombre;
    }

    public void recibeJugador(int actual, ArrayList<Jugador> todos) {
        informe(actual, todos);
    }
}